connect-inject/container_init_test.go

```diff
@@ -91,6 +91,27 @@ func TestHandlerContainerInit(t *testing.T) {
 			"",
 			`datacenter`,
 		},
+		{
+			"Prepared Query specified",
```

Comment:
Since we have two of these checks, it would be good to specify that this one checks that the `destination_type` is set correctly for a query annotation.